hips/utils/testing.py

```diff
@@ -0,0 +1,54 @@
+"""Utilities for HiPS package testing.
+
+Not of use for users / outside this package.
+"""
+import os
+from pathlib import Path
+
+import pytest
+from astropy.coordinates import SkyCoord
+
+from .wcs import WCSGeometry
+
+
+def get_hips_extra_file(filename):
+    """Get `~pathlib.Path` for a file in ``hips-extra``.
+
+    To make this work, clone this repo:
+    https://github.com/hipspy/hips-extra
+    and set an environment variable ``HIPS_EXTRA`` pointing to it.
+    """
+    path = Path(os.environ['HIPS_EXTRA'])
+    return path / filename
+
+
+def requires_hips_extra():
+    """Decorator to mark tests requiring ``hips-extra`` data.
+    """
+    has_hips_extra = get_hips_extra_file('datasets/samples/DSS2Red/properties').is_file()
```

Comment:
Locally I didn't have `HIPS_EXTRA` set, and then the tests fail like this:
```
/opt/local/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/os.py:666: in __getitem__
    value = self._data[self.encodekey(key)]
E   KeyError: b'HIPS_EXTRA'

During handling of the above exception, another exception occurred:
hips/draw/tests/test_simple.py:30: in <module>
    @requires_hips_extra()
hips/utils/testing.py:28: in requires_hips_extra
    has_hips_extra = get_hips_extra_file('datasets/samples/DSS2Red/properties').is_file()
hips/utils/testing.py:21: in get_hips_extra_file
    path = Path(os.environ['HIPS_EXTRA'])
/opt/local/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/os.py:669: in __getitem__
    raise KeyError(key) from None
E   KeyError: 'HIPS_EXTRA'
_____________________________________________________________________ ERROR collecting lib.macosx-10.12-x86_64-3.6/hips/draw/tests/test_simple.py ______________________________________________________________________
/opt/local/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/os.py:666: in __getitem__
    value = self._data[self.encodekey(key)]
E   KeyError: b'HIPS_EXTRA'

During handling of the above exception, another exception occurred:
hips/draw/tests/test_simple.py:30: in <module>
    @requires_hips_extra()
hips/utils/testing.py:28: in requires_hips_extra
    has_hips_extra = get_hips_extra_file('datasets/samples/DSS2Red/properties').is_file()
hips/utils/testing.py:21: in get_hips_extra_file
    path = Path(os.environ['HIPS_EXTRA'])
/opt/local/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/os.py:669: in __getitem__
    raise KeyError(key) from None
E   KeyError: 'HIPS_EXTRA'
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! Interrupted: 2 errors during collection !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
=============================================================================================== 2 error in 0.40 seconds ================================================================================================
```
The reason is simply that the `requires_hips_extra` implementation is buggy.
This should work:
```
def has_hips_extra():
    """Is hips-extra available? (bool)"""
    if 'HIPS_EXTRA' in os.environ:
        path = Path(os.environ['HIPS_EXTRA']) / 'datasets/samples/DSS2Red/properties'
        if path.is_file():
            return True

    return False


def requires_hips_extra():
    """Decorator to mark tests requiring ``hips-extra`` data.
    """
    skip_it = not has_hips_extra()
    reason = 'No hips-extra data available.'
    return pytest.mark.skipif(skip_it, reason=reason)
```
I didn't test that locally. @adl1995 - Please put it and you test locally that it works as expected by running the tests with and without having `HIPS_EXTRA` env variable set.